x-pack/plugins/security/server/authentication/providers/saml.ts

```diff
@@ -404,6 +362,43 @@ export class SAMLAuthenticationProvider extends BaseAuthenticationProvider {
         ? AuthenticationResult.notHandled()
         : AuthenticationResult.failed(err);
     }
+
+    // Now we need to retrieve full user information.
+    let user: Readonly<AuthenticatedUser>;
```

Comment:
We can definitely do that if it would help.

Reply:
That'd be awesome! It's not critical since this additional round-trip is done once per session, but it'd be nice to eliminate that nevertheless assuming it makes sense to you as well. Currently we have to additionally fetch user info after the following API calls that return access and optionally refresh tokens:

* `/_security/oauth2/token` (Kerberos and Token)
* `/_security/delegate_pki` (PKI)
* `/_security/saml/authenticate` (SAML)
* `/_security/oidc/authenticate` (OIDC)

I'll file the discuss issue in ES repository and we can see if/where we can attach user info.

Reply:
Filed: https://github.com/elastic/elasticsearch/issues/59685